interactive/dashboard/upload_handler.py

```diff
@@ -114,21 +118,29 @@ def copyConfigFile(source, destination):
 
 
 # loads uploaded zip configuration
-def handle_uploaded_zip_config(f):
+def handle_uploaded_zip_config(f, uploaded_path="dashboard/static/zip", 
+                               config_path=os.path.join(settings.BASE_DIR, "dashboard/static/config")):
     content = f.read()
-    file_name = os.path.join(os.path.join(settings.BASE_DIR, "dashboard/static/zip/uploaded"), 'uploaded.zip')
+    file_name = os.path.join(os.path.join(settings.BASE_DIR, uploaded_path+"/uploaded"), 'uploaded.zip')
+    if os.path.isdir(file_name.replace('uploaded.zip', '')) == False:
+        # make dirs just in case
+        os.makedirs(file_name.replace('uploaded.zip', ''))
+    if os.path.isfile(file_name) == False:
+        
+        print("* created directory for uploaded zip file:", file_name.replace('uploaded.zip', ''))
+        g = open(file_name, 'x')
     g = open(file_name, 'wb')
     g.write(content)
     g.close()
```

Comment:
So we could open the file twice? What's going on here?

Reply:
if the zip doesn't exist we have to create it first, unfortunately u can't do w+ for zip files, it's very weird